Sources/TuistKit/Services/TestService.swift

```diff
@@ -276,72 +279,84 @@ final class TestService { // swiftlint:disable:this type_body_length
-                graph: graph
-            )
+        try await testSchemes(
+            schemes,
+            graph: graph,
+            mapperEnvironment: mapperEnvironment,
+            cacheStorage: cacheStorage,
+            clean: clean,
+            configuration: configuration,
+            version: version,
+            deviceName: deviceName,
+            platform: platform,
+            rosetta: rosetta,
+            resultBundlePath: resultBundlePath,
+            derivedDataPath: derivedDataPath,
+            retryCount: retryCount,
+            testTargets: testTargets,
+            skipTestTargets: skipTestTargets,
+            testPlanConfiguration: testPlanConfiguration,
+            passthroughXcodeBuildArguments: passthroughXcodeBuildArguments
+        )
+    }
 
-            for testScheme in testSchemes {
-                try await self.testScheme(
-                    scheme: testScheme,
-                    graphTraverser: graphTraverser,
-                    clean: clean,
-                    configuration: configuration,
-                    version: version,
-                    deviceName: deviceName,
-                    platform: platform,
-                    rosetta: rosetta,
-                    resultBundlePath: resultBundlePath,
-                    derivedDataPath: derivedDataPath,
-                    retryCount: retryCount,
-                    testTargets: testTargets,
-                    skipTestTargets: skipTestTargets,
-                    testPlanConfiguration: testPlanConfiguration,
-                    passthroughXcodeBuildArguments: passthroughXcodeBuildArguments
-                )
-            }
-        } else {
-            let allSchemes = buildGraphInspector.workspaceSchemes(graphTraverser: graphTraverser)
-            testSchemes = allSchemes
-                .filter {
-                    $0.testAction.map { !$0.targets.isEmpty } ?? false
-                }
+    // MARK: - Helpers
 
-            if testSchemes.isEmpty {
-                logger.log(level: .info, "There are no tests to run, finishing early")
-                return
+    private func testSchemes(
+        _ schemes: [Scheme],
+        graph: Graph,
+        mapperEnvironment: MapperEnvironment,
+        cacheStorage: CacheStoring,
+        clean: Bool,
+        configuration: String?,
+        version: Version?,
+        deviceName: String?,
+        platform: String?,
+        rosetta: Bool,
+        resultBundlePath: AbsolutePath?,
+        derivedDataPath: AbsolutePath?,
+        retryCount: Int,
+        testTargets: [TestIdentifier],
+        skipTestTargets: [TestIdentifier],
+        testPlanConfiguration: TestPlanConfiguration?,
+        passthroughXcodeBuildArguments: [String]
+    ) async throws {
+        let graphTraverser = GraphTraverser(graph: graph)
+        let testSchemes = schemes
+            .filter {
+                $0.testAction.map { !$0.targets.isEmpty } ?? false
```

Comment:
Does this account for test plans too?

Reply:
If the scheme ends up with no targets, then so should the action. I added a test we report the numbers well for the action plan.